syntax = "proto3";
package lastmile.v1;
option go_package = "github.com/yourorg/lastmile/api/gen/go/lastmile/v1;lastmilev1";
import "lastmile/v1/common.proto";

service StationService {
  rpc UpsertStation(UpsertStationRequest) returns (UpsertStationResponse);
  rpc GetStation(GetStationRequest) returns (GetStationResponse);
  rpc ListStations(Empty) returns (ListStationsResponse);
  rpc NearbyAreas(GetStationRequest) returns (NearbyAreasResponse);
}

message UpsertStationRequest { Station station = 1; }
message UpsertStationResponse { Station station = 1; }
message GetStationRequest { string id = 1; }
message GetStationResponse { Station station = 1; }
message ListStationsResponse { repeated Station stations = 1; }
message NearbyAreasResponse { repeated string nearby_areas = 1; }
